src/bentoml/_internal/cloud/client.py

```diff
@@ -49,7 +49,11 @@ def __init__(self, endpoint: str, api_token: str) -> None:
 
     def _is_not_found(self, resp: requests.Response) -> bool:
         # Forgive me, I don't know how to map the error returned by gorm to juju/errors
-        return resp.status_code == 400 and "record not found" in resp.text
+        return (
+            resp.status_code == 404
+            or resp.status_code == 400
```

Comment:
Why are we return of it is 404?


Ah I was always meant to submit a PR for error improvement on this end. Eh maybe when we refactor cloud_rest_client

Reply:
I found that now when the model or deployment doesn't exist, it returns 404.

Reply:
This is changed by me, we have fixed the error handling of BentoCloud that correctly returns 404 for not found.